src/uuid.rs

```diff
@@ -0,0 +1,32 @@
+use failure::Error;
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct RawUuid<T>(T);
+
+#[cfg(feature = "safe-uuid")]
+impl RawUuid<::uuid::Uuid> {
+    pub(crate) fn to_string(&self) -> String {
+        self.0.to_string()
+    }
+
+    pub(crate) fn parse_str(input: &str) -> Result<Self, Error> {
+        Ok(Self(uuid::Uuid::parse_str(input)?))
+    }
+}
+
+#[cfg(not(feature = "safe-uuid"))]
```

Comment:
This was so that folks who might want to use a different bignum or uuid library could do so if they desired. However, I think it's probably more prudent to address that case if it ever comes up rather than add more code for a case we don't know exists. I'll make the safe version the only implementation for now and we can address the raw case if it ever comes up.